aten/src/ATen/native/mps/operations/SummaryOps.mm

```diff
@@ -0,0 +1,146 @@
+                                                   shape:getMPSShape(self)
+                                                dataType:getMPSDataType(output.scalar_type())];
+          }
+
+          MPSGraphTensor *outputTensor = [mpsGraph scatterWithDataTensor:scatterDataTensor
+                                                           updatesTensor:updatesTensor
+                                                           indicesTensor:inputTensor
+                                                                     axis:0
+                                                                     mode:MPSGraphScatterModeAdd
+                                                                     name:nil];
+
+          newCachedGraph->inputTensor_ = inputTensor;
+          newCachedGraph->outputTensor_ = outputTensor;
+          newCachedGraph->scatterDataTensor_ = scatterDataTensor;
+          if (has_weights) {
+            newCachedGraph->weightsTensor_ = updatesTensor;
+          }
+        }
+        return newCachedGraph;
+      });
+      cachedGraph = static_cast<CachedGraph *>(tmpCachedGraph);
+    }
+
+    // Create placeholders which use the keys of the CachedGraph to create inputs and outputs of the operation
+    Placeholder inputPlaceholder = Placeholder(cachedGraph->inputTensor_, self);
+    Placeholder outputPlaceholder = Placeholder(cachedGraph->outputTensor_, output);
+    Placeholder scatterPlaceholder = Placeholder(cachedGraph->scatterDataTensor_, output);
+    Placeholder weightsPlaceholder = Placeholder();
+
+    // Create dictionary of inputs/feeds and outputs/results
+    NSMutableDictionary<MPSGraphTensor*, MPSGraphTensorData*>* feeds =[NSMutableDictionary dictionary];
+    feeds[inputPlaceholder.getMPSGraphTensor()] = inputPlaceholder.getMPSGraphTensorData();
+    feeds[scatterPlaceholder.getMPSGraphTensor()] = scatterPlaceholder.getMPSGraphTensorData();
+    if(has_weights) {
+      weightsPlaceholder = Placeholder(cachedGraph->weightsTensor_, weights);
+      feeds[weightsPlaceholder.getMPSGraphTensor()] = weightsPlaceholder.getMPSGraphTensorData();
+    }
+
+    NSDictionary<MPSGraphTensor*, MPSGraphTensorData*>* results = @{
+      outputPlaceholder.getMPSGraphTensor() : outputPlaceholder.getMPSGraphTensorData()
+    };
+
+    // Run the graph
+    runMPSGraph(stream, cachedGraph->graph(), feeds, results);
+  }
+
+  return output;
+}
+
+Tensor _bincount_mps(const Tensor& self, const c10::optional<Tensor>& weights_opt, int64_t minlength) {
+  // See [Note: hacky wrapper removal for optional tensor]
+  c10::MaybeOwned<Tensor> weights_maybe_owned = at::borrow_from_optional_tensor(weights_opt);
+  const Tensor& weights = *weights_maybe_owned;
+
+  TORCH_CHECK(c10::isIntegralType(self.scalar_type(), /*includesBool=*/true));
+  TORCH_CHECK(minlength >= 0, "minlength should be >= 0");
+
+  if (self.dim() == 1 && self.numel() == 0) {
+    return at::zeros(
+        {minlength},
+        kLong,
```

Comment:
Is the return type always kLong? Couldn't it be self.scalar_type()?

Reply:
Yes, in case of empty input tensor, it should always return a tensor of size `minlength` filled with zeros. When there are weights passed in and the input is not empty, then output will use the same dtype as the weights. 

> "If minlength is specified, the number of bins is at least minlength and if input is empty, then the result is tensor of size minlength filled with zeros".